bcda/api/v1/api.go

```diff
@@ -293,20 +301,75 @@ func ServeData(w http.ResponseWriter, r *http.Request) {
 			break
 		}
 	}
-
+	w.Header().Set(constants.ContentType, "application/fhir+ndjson")
 	if useGZIP {
 		w.Header().Set("Content-Encoding", "gzip")
 		gz := gzip.NewWriter(w)
 		defer gz.Close()
 
 		gzw := gzipResponseWriter{Writer: gz, ResponseWriter: w}
```

Comment:
Not sure how heavy of a process this is, did we just want to add it to the encoded-else check?

Reply:
I was thinking that the scope of isGzipEncoded would be just to determine if the magic number of a file (if it can be read) implies the file is gzip-encoded, and then the actual opening / writing of the file could be handled within the main function utilizing it. 
Only one of the four paths actually opens the file again (rather than using a writer and the http.serveFile method), so I was thinking that may be an acceptable tradeoff rather than having to allocate memory for a path that will be chosen in a small minority of situations (data encoded + no gzip header, which right now is <5% of cases).

Reply:
Ah okay, thanks for the explanation!